### PR TITLE
CI/CD with ARM templates for TDI system

### PR DESCRIPTION
**Pull request by kolendomichal**
Yaml template for building, running tests , creating resource group and deploying webapp application
fixes #AB54

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,34 +1,102 @@
-# ASP.NET
-# Build and test ASP.NET projects.
-# Add steps that publish symbols, save build artifacts, deploy, and more:
-# https://docs.microsoft.com/azure/devops/pipelines/apps/aspnet/build-aspnet-4
-
 trigger:
 - main
```
Comment (michalchecinski) [inline azure-pipelines.yml]: We've got `master` branch :)

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,34 +1,102 @@
...
+resources:
+- repo: self
+
 variables:
-  solution: '**/*.sln'
-  buildPlatform: 'Any CPU'
-  buildConfiguration: 'Release'
+  - group: tdi-base
+  - name: solution
+    value: '**/*.sln'
```
Comment (michalchecinski) [inline azure-pipelines.yml]: Just a suggestion: Maybe specify a solution name?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -1,34 +1,102 @@
-# ASP.NET
-# Build and test ASP.NET projects.
-# Add steps that publish symbols, save build artifacts, deploy, and more:
-# https://docs.microsoft.com/azure/devops/pipelines/apps/aspnet/build-aspnet-4
-
 trigger:
```
Comment (michalchecinski) [inline azure-pipelines.yml:1]: Maybe add a `batch: true` statement, and exclude `README` and `images/` folder from build trigger?

File: infrastructure/template.json [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        "components_TechnicalDocuIndexerWeb_name": {
+            "defaultValue": "TechnicalDocuIndexerWeb",
+            "type": "String"
+        }
+    },
+    "variables": {},
+    "resources": [
+        {
+            "type": "microsoft.insights/components",
+            "apiVersion": "2018-05-01-preview",
```
Comment (michalchecinski) [inline infrastructure/template.json]: Why using preview api version?

File: infrastructure/template.json [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+            "location": "West Europe",
+            "dependsOn": [
+                "[resourceId('Microsoft.Web/serverfarms', parameters('serverfarms_ASP_test_bfbf_name'))]"
+            ],
+            "kind": "app",
+            "properties": {
+                "enabled": true,
+                "hostNameSslStates": [
+                    {
+                        "name": "TechnicalDocuIndexerWeb.azurewebsites.net",
```
Comment (michalchecinski) [inline infrastructure/template.json]: Parametrize this url

File: infrastructure/template.json [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+                        "sslState": "Disabled",
+                        "hostType": "Standard"
+                    },
+                    {
+                        "name": "TechnicalDocuIndexerWeb.scm.azurewebsites.net",
```
Comment (michalchecinski) [inline infrastructure/template.json]: Parametrize this url

File: src/TechnicalDocuIndexer.Web/pipelines/ci-cd-tdi-deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
+﻿trigger:
```
Comment (michalchecinski) [inline src/TechnicalDocuIndexer.Web/pipelines/ci-cd-tdi-deployment.yaml]: Why is this file? :)

File: infrastructure/parameters.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (michalchecinski) [inline infrastructure/parameters.json:1]: Overall suggestion: use the Azure naming conventions in naming:
https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-naming
https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations

File: infrastructure/parameters.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "serverfarms_ASP_test_bfbf_name": {
+            "value": "ASP-test-bfbf"
```
Comment (michalchecinski) [inline infrastructure/parameters.json]: especially change this name 😄

File: infrastructure/template.json [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+            "type": "String"
+        }
+    },
+    "variables": {},
+    "resources": [
+        {
+            "type": "microsoft.insights/components",
+            "apiVersion": "2018-05-01-preview",
+            "name": "[parameters('components_TechnicalDocuIndexerWeb_name')]",
+            "location": "westeurope",
```
Comment (michalchecinski) [inline infrastructure/template.json]: Location should be parametrized as a variable for example: `"location": "[resourceGroup().location]"`

File: infrastructure/template.json [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+                "publicNetworkAccessForIngestion": "Enabled",
+                "publicNetworkAccessForQuery": "Enabled"
+            }
+        },
+        {
+            "type": "Microsoft.Web/serverfarms",
+            "apiVersion": "2018-02-01",
+            "name": "[parameters('serverfarms_ASP_test_bfbf_name')]",
+            "location": "West Europe",
+            "sku": {
```
Comment (michalchecinski) [inline infrastructure/template.json:40]: I would make SKU as a parameter 😃

**Review by michalchecinski — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml (hunk 1 above)
Comment (kolendomichal) [reply azure-pipelines.yml]: fixed

File: azure-pipelines.yml (hunk 3 above)
Comment (kolendomichal) [reply azure-pipelines.yml:1]: noted, added

File: infrastructure/template.json (hunk 5 above)
Comment (kolendomichal) [reply infrastructure/template.json]: parametrized

File: azure-pipelines.yml (hunk 2 above)
Comment (kolendomichal) [reply azure-pipelines.yml]: specified solution name

File: src/TechnicalDocuIndexer.Web/pipelines/ci-cd-tdi-deployment.yaml (hunk 7 above)
Comment (kolendomichal) [reply src/TechnicalDocuIndexer.Web/pipelines/ci-cd-tdi-deployment.yaml]: mistake, removed

File: infrastructure/parameters.json (hunk 9 above)
Comment (kolendomichal) [reply infrastructure/parameters.json]: changed

File: infrastructure/template.json (hunk 10 above)
Comment (kolendomichal) [reply infrastructure/template.json]: parametrized location

File: infrastructure/template.json (hunk 11 above)
Comment (kolendomichal) [reply infrastructure/template.json:40]: I understand the need, but we will not make it more than the Free tier, as to not waste $$.
Nonetheless in the normal project, I would make it as parameters

File: infrastructure/template.json (hunk 4 above)
Comment (kolendomichal) [reply infrastructure/template.json]: fixed

File: infrastructure/parameters.json (hunk 8 above)
Comment (kolendomichal) [reply infrastructure/parameters.json:1]: will study it, too lazy to fix now 🥇

**Pull request merged**